lib/spack/spack/relocate.py

```diff
@@ -413,92 +495,75 @@ def replace(match):
             return data
         return match.group().replace(old_dir.encode('utf-8'),
                                      new_dir.encode('utf-8')) + b'\0' * padding
+    if len(new_dir) > len(old_dir):
+        err_msg = "The new path %s" % new_dir
```

Comment:
OK I will add a debug message. I had some debate with Chris Green about this should be a warning or a fatal exception. For mach-o binaries you cannot do the replacement under these conditions. Since I refined the library id and dependencies and macholib can handle the replacement under these conditions, I think skipping the rpath replacements should just be a warning. I never ran into the case where a compiled in path replacement was needed.